src/NHibernate/Dialect/Schema/AbstractDataBaseSchema.cs

```diff
@@ -99,6 +101,16 @@ public virtual ISet<string> GetReservedWords()
 			{
 				result.Add(row["ReservedWord"].ToString());
 			}
+
+			if (IncludeDataTypesInReservedWords)
+			{
+				DataTable dtTypes = connection.GetSchema(DbMetaDataCollectionNames.DataTypes);
+				foreach (DataRow row in dtTypes.Rows)
```

Comment:
Anywhere 17 supports this too, without a different name (using `TypeName` works for it). But its schema specific class overrides that method, and has not be updated for including data types. Maybe should it be updated.